From 019422d19c6401fa8f45b1311ecd31bb087c809d Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 17 Jul 2018 10:13:38 +0800
Subject: [PATCH] conditionally do not fetch code by easy_install

If var-NO_FETCH_BUILD is set, do not allow to fetch code from
internet by easy_install.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 setuptools/command/easy_install.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/setuptools/command/easy_install.py b/setuptools/command/easy_install.py
index 85ee40f..7eb9d2a 100755
--- a/setuptools/command/easy_install.py
+++ b/setuptools/command/easy_install.py
@@ -637,6 +637,11 @@ class easy_install(Command):
             os.path.exists(tmpdir) and rmtree(rmtree_safe(tmpdir))
 
     def easy_install(self, spec, deps=False):
+        if os.environ.get('NO_FETCH_BUILD', None):
+            log.error("ERROR: Do not try to fetch `%s' for building. "
+                      "Please add its native recipe to DEPENDS." % spec)
+            return None
+
         if not self.editable:
             self.install_site_py()
 
-- 
2.7.4

